From 8b71889044aba0065b957f2145318640ce73f45d Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Fri, 9 Nov 2018 15:16:23 +0800
Subject: [PATCH 5073/5242] MLK-20247: ASoC: imx-rpmsg: fix error when m4
 image is not loaded

commit  532a86fb069c52a824d70060bc6b444b6eff2eb8 from
https://source.codeaurora.org/external/imx/linux-imx.git

The reason is same as commit d4eb8ab26399 ("MLK-19854-1: ASoC:
imx-cs42888: fix error when m4 image is not loaded")

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-rpmsg.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/soc/fsl/imx-rpmsg.c b/sound/soc/fsl/imx-rpmsg.c
index 8bbe044..c782fb9 100644
--- a/sound/soc/fsl/imx-rpmsg.c
+++ b/sound/soc/fsl/imx-rpmsg.c
@@ -35,6 +35,8 @@ static int imx_rpmsg_probe(struct platform_device *pdev)
 	struct platform_device *cpu_pdev;
 	struct imx_rpmsg_data *data;
 	struct fsl_rpmsg_i2s         *rpmsg_i2s;
+	struct snd_soc_dai_link_component dlc = { 0 };
+	struct snd_soc_dai *codec_dai;
 	int ret;
 
 	cpu_np = of_parse_phandle(pdev->dev.of_node, "cpu-dai", 0);
@@ -83,6 +85,12 @@ static int imx_rpmsg_probe(struct platform_device *pdev)
 			    SND_SOC_DAIFMT_CBS_CFS;
 	}
 
+	dlc.name = data->dai[0].codec_name;
+	dlc.dai_name = data->dai[0].codec_dai_name;
+	codec_dai = snd_soc_find_dai(&dlc);
+	if (!codec_dai)
+		return -ENODEV;
+
 	data->dai[0].cpu_dai_name = dev_name(&cpu_pdev->dev);
 	data->dai[0].platform_of_node = cpu_np;
 	data->dai[0].playback_only = true;
-- 
1.7.9.5

